src/test/java/com/google/roomies/database/FirebaseDatabaseTest.java

```diff
@@ -99,6 +111,38 @@ public void testAddListingAsMap_addsSingleListingToFirestore() throws Exception
     verify(collectionMock, Mockito.times(1)).add(listingData);
   }
 
+  @Test
+  public void testaddCommentAsMapToListing_addsSingleCommentToFirestore() throws Exception {
+    when(docSnapshotMock.exists()).thenReturn(true);
+    String listingId = "7YDcsjQOTzVoUxeXiysT";
+    comment = Comment.builder()
+      .setComment("Test comment")
+      .build();
+    ImmutableMap<String, Object> commentData = comment.toMap();
+    when(collectionMock.add(commentData)).thenReturn(docReferenceFutureMock);
+    when(docReferenceMock.collection(COMMENT_COLLECTION_NAME)).thenReturn(collectionMock);
+
+    database.addCommentAsMapToListing(comment, listingId);
+
+    verify(collectionMock, Mockito.times(1)).add(commentData);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testaddCommentAsMapToListing_listingIdIsInvalid_throwsIllegalArgumentException()
+      throws InterruptedException, ExecutionException {
+    when(docSnapshotMock.exists()).thenReturn(false);
+    String listingId = "7YDcsjQOTzVoUxeXiysT";
+    String commentText = "Test comment";
```

Comment:
Define it as a final on class level, rather than creating in every test. You can also be build Comment as private static final.

Reply:
Changed comment parameters and Comment instance to private static final variables.